src/main/java/org/clang/struct/CXCursor.java

```diff
@@ -88,6 +90,16 @@ public CXSourceLocation getCursorLocation() {
         return clang.clang_getCursorLocation(this);
     }
 
+    public CXPlatformAvailability[] getCursorPlatformAvailabilities() {
+        // I identified 5 different platforms, just for safety we fetch a maximum of 6
+        int size = 6;
+        Ptr<CXPlatformAvailability> availabilityPtr = PtrFactory.newStructArray(CXPlatformAvailability.class, size);
+        int i = clang.clang_getCursorPlatformAvailability(this, null, null, null, null, availabilityPtr, size);
+        CXPlatformAvailability[] availabilities = new CXPlatformAvailability[i];
+        availabilityPtr.copyTo(availabilities, i);
```

Comment:
According to https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__MANIP.html#gaab07659398c4538771d62c81ca5dea69

This function returns

> The number of platforms (N) for which availability information is available (which is unrelated to availability_size).

So here we should use `min(i, size)` instead.

Reply:
Or, even better, call `clang_getCursorPlatformAvailability` with all `null`s, then use the returned value as the size and call this function again.

Reply:
Good idea! Done!